src/test/java/seedu/address/logic/commands/RemoveCommandTest.java

```diff
@@ -0,0 +1,109 @@
+        Deadline deadlineToDelete = model.getFilteredDeadlineList().get(INDEX_FIRST_DEADLINE.getZeroBased());
+        RemoveCommand removeCommand = new RemoveCommand(INDEX_FIRST_DEADLINE);
+
+        String expectedMessage = String.format(RemoveCommand.MESSAGE_DELETE_DEADLINE_SUCCESS, deadlineToDelete);
+
+        ModelManager expectedModel = new ModelManager(model.getAddressBook(), new UserPrefs());
+        expectedModel.deleteDeadline(deadlineToDelete);
+
+        assertCommandSuccess(removeCommand, model, expectedMessage, expectedModel);
+    }
+
+    @Test
+    public void execute_invalidIndexUnfilteredList_throwsCommandException() {
+        Index outOfBoundIndex = Index.fromOneBased(model.getFilteredDeadlineList().size() + 1);
+        RemoveCommand removeCommand = new RemoveCommand(outOfBoundIndex);
+
+        assertCommandFailure(removeCommand, model, Messages.MESSAGE_INVALID_DEADLINE_DISPLAYED_INDEX);
+    }
+
+    @Test
+    public void execute_validIndexFilteredList_success() {
+        showDeadlineAtIndex(model, INDEX_FIRST_DEADLINE);
+
+        Deadline deadlineToDelete = model.getFilteredDeadlineList().get(INDEX_FIRST_DEADLINE.getZeroBased());
+        RemoveCommand removeCommand = new RemoveCommand(INDEX_FIRST_DEADLINE);
+
+        String expectedMessage = String.format(RemoveCommand.MESSAGE_DELETE_DEADLINE_SUCCESS, deadlineToDelete);
+
+        Model expectedModel = new ModelManager(model.getAddressBook(), new UserPrefs());
+        expectedModel.deleteDeadline(deadlineToDelete);
+        showNoPerson(expectedModel);
+
+        assertCommandSuccess(removeCommand, model, expectedMessage, expectedModel);
+    }
+
+    @Test
+    public void execute_invalidIndexFilteredList_throwsCommandException() {
+        showDeadlineAtIndex(model, INDEX_FIRST_DEADLINE);
+
+        Index outOfBoundIndex = INDEX_SECOND_DEADLINE;
+        // ensures that outOfBoundIndex is still in bounds of address book list
+        assertTrue(outOfBoundIndex.getZeroBased() < model.getAddressBook().getDeadlineList().size());
+
+        RemoveCommand removeCommand = new RemoveCommand(outOfBoundIndex);
+
+        assertCommandFailure(removeCommand, model, Messages.MESSAGE_INVALID_DEADLINE_DISPLAYED_INDEX);
+    }
+
+    @Test
+    public void equals() {
+        RemoveCommand removeFirstCommand = new RemoveCommand(INDEX_FIRST_DEADLINE);
+        RemoveCommand removeSecondCommand = new RemoveCommand(INDEX_SECOND_DEADLINE);
+
+        // same object -> returns true
+        assertTrue(removeFirstCommand.equals(removeFirstCommand));
+
+        // same values -> returns true
+        RemoveCommand removeFirstCommandCopy = new RemoveCommand(INDEX_FIRST_DEADLINE);
+        assertTrue(removeFirstCommand.equals(removeFirstCommandCopy));
+
+        // different types -> returns false
+        assertFalse(removeFirstCommand.equals(1));
+
+        // null -> returns false
+        assertFalse(removeFirstCommand.equals(null));
```

Comment:
`anyObject.equals(null)` can only return false or throw a `NullPointerException`. But then again we're testing the `equals()` method ah, so I'm not v sure how to go about this LOL